doc/modules/array_api.rst

```diff
@@ -219,3 +221,13 @@ certain combinations of array namespaces and devices, such as `PyTorch on MPS`
 scikit-learn will revert to using the `float32` data type instead. This can result in
 different behavior (typically numerically unstable results) compared to not using array
 API dispatching or using a device with `float64` support.
+
+.. _array_api_support_for_convenience:
+
+Note on functions supporting array api only for convenience
+-----------------------------------------------------------
+
+Certain functions within scikit-learn only support the array api for convenience
+purposes but internally the arrays are converted to numpy arrays for performing
+the required underlying computations and then converted back to the array
+namespace under consideration (e.g., :func:`metrics.pairwise.manhattan_distances`).
```

Comment:
Maybe we can document the list here.

Reply:
+1 for maintaining a list of estimators that accept array API inputs by internally converting to NumPy/CPU).